src/components/PathwaysList/PathwaysList.tsx

```diff
@@ -36,19 +37,43 @@ const PathwaysList: FC<PathwaysListProps> = ({
   service,
   resources
 }) => {
+  const [criteria, setCriteria] = useState<CriteriaResult[] | null>(null);
+
+  if (!criteria && evaluatedPathways.length > 0 && resources != null && resources.length > 0) {
+    // Create a fake Bundle for the CQL engine and check if patientPath needs to be evaluated
+    const patient = {
+      resourceType: 'Bundle',
+      entry: resources.map((r: fhir.Resource) => ({ resource: r }))
+    };
+
+    // Evaluate pathway criteria for each pathway
+    const criteriaPromises: Promise<CriteriaResult>[] = [];
+    evaluatedPathways.forEach(pathway => {
+      criteriaPromises.push(evaluatePathwayCriteria(patient, pathway.pathway));
+    });
+    Promise.all(criteriaPromises).then(criteriaResults => {
+      setCriteria(criteriaResults.sort((a, b) => b.matches - a.matches));
+    });
+  }
+
   function renderList(): ReactNode {
     return (
       <div>
-        {evaluatedPathways.map(evaluatedPathway => {
-          return (
-            <PathwaysListElement
-              evaluatedPathway={evaluatedPathway}
-              callback={callback}
-              resources={resources}
-              key={evaluatedPathway.pathway.name}
-            />
-          );
-        })}
+        {criteria &&
+          criteria.map(c => {
+            const evaluatedPathway = evaluatedPathways.find(p => p.pathway.name === c.pathwayName);
+            if (evaluatedPathway)
+              return (
+                <PathwaysListElement
+                  evaluatedPathway={evaluatedPathway}
+                  callback={callback}
+                  resources={resources}
+                  criteria={c}
+                  key={evaluatedPathway.pathway.name}
+                />
+              );
+            else return <div></div>;
```

Comment:
Could we put something here as a loading indicator? Nothing too crazy, just something that says "yes the system is working to load a pathway" when `criteria` is null. It will be especially nice on the demo endpoint when things move slowly.

Reply:
So I can certainly add that functionality but this else is for when the app cannot find a matching pathway for the criteria result. It should never happen but I had to put the else in